pystac/extensions/projection.py

```diff
@@ -477,8 +477,7 @@ def migrate(
             return
 
         # proj:epsg moved to proj:code
-        if "proj:epsg" in obj["properties"]:
-            epsg = obj["properties"]["proj:epsg"]
+        if epsg := obj["properties"].get("proj:epsg"):
             obj["properties"]["proj:code"] = f"EPSG:{epsg}"
             del obj["properties"]["proj:epsg"]
```

Comment:
I think we still want to delete the key. Maybe 
```suggestion
        if epsg := obj["properties"].pop("proj:epsg", None):
            obj["properties"]["proj:code"] = f"EPSG:{epsg}"
```